wrap.sh

```diff
@@ -178,26 +179,42 @@ while getopts "r:w:e:abcdhmnpuv" opt; do
     env_vars+=(DBUS_SESSION_BUS_ADDRESS)
     ;;
 
-  # grant desktop access, wayland, X11, DRI
+  # grant desktop access (Wayland or X11) and rendering hardware access
   d)
-    bwrap_opts+=(--bind "$XDG_RUNTIME_DIR/$WAYLAND_DISPLAY" "$XDG_RUNTIME_DIR/$WAYLAND_DISPLAY")
+    if [ -n "${WAYLAND_DISPLAY:-}" ] && [ -n "${XDG_RUNTIME_DIR:-}" ]; then
+      # Using Wayland: bind the Wayland display socket
+      bwrap_opts+=(--bind "$XDG_RUNTIME_DIR/$WAYLAND_DISPLAY" "$XDG_RUNTIME_DIR/$WAYLAND_DISPLAY")
+    elif [ -n "${DISPLAY:-}" ]; then
+      # Using X11: bind the X11 socket directory
+      # The standard location is usually /tmp/.X11-unix.
+      if [ -d "/tmp/.X11-unix" ]; then
+        bwrap_opts+=(--ro-bind "/tmp/.X11-unix" "/tmp/.X11-unix")
+      fi
+
+      # Bind the .Xauthority file so that the authorization data is available.
+      if [ -n "XAUTHORITY" ]; then
+        bwrap_opts+=(--ro-bind "$XAUTHORITY" "$HOME/.Xauthority")
+        env_vars+=(XAUTHORITY)
+      fi
+    fi
 
-    if [ -f /dev/dri ]; then
+    if [ -e /dev/dri ]; then
```

Comment:
```suggestion
    if [ -d /dev/dri ]; then
```
Good catch, maybe we even want to put `-d`?

`man test`:
```
       -d FILE
              FILE exists and is a directory

       -e FILE
              FILE exists

       -f FILE
              FILE exists and is a regular file
```